### PR TITLE
Netcdf4 export3

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

See pull #401, provides a warning to user when metadata is supplied that could cause unexpected behavior when written to a netCDF4 file.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally when producing internal files for satellite missions.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
My work translating the commits from #401 was clearly less than ideal. That being said, those other commits already in the system so actual harm done. I will try to clean up the sw_f107.py changes.

File: pysat/_instrument.py [hunk 1]
```diff
@@ -2017,12 +2017,29 @@ def _filter_netcdf4_metadata(self, mdata_dict, coltype, remove=False):
         if (coltype == type(' ')) or (coltype == type(u' ')):
             # if isinstance(coltype, str):
             remove = True
+            warnings.warn('FillValue is not an acceptable '
```
Comment (aburrell) [inline pysat/_instrument.py:2020]: `warn` has been deprecated, use `warning`

Comment (aburrell) [reply pysat/_instrument.py:2020]: Here and elsewhere.

Comment (rstoneback) [reply pysat/_instrument.py]: Thanks for the note. Updated.

**Comment by rstoneback** [comment]
> It looks like the f017 bug fix is in this pull. Is this what you were referring to on the call? 
Yes.

> If so, can you also update the changes to the comments from #399 here?
Yes.

**Comment by jklenzing** [comment]
I think `warn` is deprecated in logging, not warning.  Which one should we be using?

**Comment by rstoneback** [comment]
> I think `warn` is deprecated in logging, not warning. Which one should we be using?

I just do what I'm told :)
Looks like logging. Thanks Jeff.

**Comment by jklenzing** [comment]
Changelog handled in 2.2

**Comment by rstoneback** [comment]
Thanks for checking.

**Comment by rstoneback** [comment]
Adding note that pull #427 should be incorporated first.

**Comment by jklenzing** [comment]
Support for basestr was removed in #438.  Updated for compatibility post merge.

**Comment by jklenzing** [comment]
Upated description in #438 to include basestr removal.  This branch is working again.  Any other changes needed, or is it ready for review?

**Comment by rstoneback** [comment]
It is ready!

File: pysat/_instrument.py [hunk 2]
```diff
@@ -1988,12 +1991,29 @@ def _filter_netcdf4_metadata(self, mdata_dict, coltype, remove=False,
...
+        
         # print('coltype', coltype, remove, type(coltype), )
         if u'_FillValue' in mdata_dict.keys():
             # make sure _FillValue is the same type as the data
             if remove:
                 mdata_dict.pop('_FillValue')
             else:
+                if not np.can_cast(mdata_dict['_FillValue'], coltype):
+                    if 'FieldNam' in mdata_dict:
+                         warnings.warn('FillValue for %s (%s) cannot be safely '
```
Comment (jklenzing) [inline pysat/_instrument.py:2005]: indentation

File: pysat/instruments/sw_f107.py [hunk 3]
```diff
@@ -676,10 +675,6 @@ def rewrite_daily_file(year, outfile, lines):
 
     """
 
-    # Parse text to get the date the prediction was generated
-    date_str = lines.split(':Issued: ')[-1].split('\n')[0]
-    date = dt.datetime.strptime(date_str, '%H%M UT %d %b %Y')
-
```
[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/instruments/sw_f107.py:682]: Perhaps you are remembering the develop-2 pull version?

[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/instruments/sw_f107.py:682]: I'm not sure why this change ended up here exactly, but the code is in develop-3.

**Comment by rstoneback** [comment]
@jklenzing I've updated this pull - fyi in case you'd like to take a gander before I merge it in.

File: pysat/_instrument.py [hunk 4]
```diff
@@ -11,6 +11,8 @@
 import sys
 import warnings
 
+import netCDF4
+import numpy as np
```
[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/_instrument.py:14]: I removed the earlier one.

**Comment by jklenzing** [comment]
Looks good!

**Pull request merged**